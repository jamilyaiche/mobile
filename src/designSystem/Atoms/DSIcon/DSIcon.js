import PropTypes from 'prop-types'
import React from 'react'
import { TouchableOpacity } from 'react-native'
import { createIconSetFromFontello } from 'react-native-vector-icons'
import fontelloConfig from '../../../assets/fontIconsConfig/config.json'
import Colors from '../../../assets/styles/Colors'

const Icon = createIconSetFromFontello(fontelloConfig)

const DSIcon = (props) => {
  let size = 20
  if (props.size === 'xxx-small') {
    size = 5
  }
  if (props.size === 'extra-small') {
    size = 14
  }
  if (props.size === 'medium') {
    size = 40
  }
  if (props.size === 'large') {
    size = 60
  }
  if (props.size === 'extra-large') {
    size = 80
  }
  return (
    <TouchableOpacity
      {...(props.onLongPress !== undefined && { onLongPress: props.onLongPress })}
      {...(props.onPress !== undefined && { onPress: props.onPress })}
      {...(props.onPressIn !== undefined && { onPressIn: props.onPressIn })}
      {...(props.onPressOut !== undefined && { onPressOut: props.onPressOut })}
      {...(props.style !== undefined && { style: [props.style, { padding: 1 }] })}
      {...(props.disabled !== undefined && { disabled:props.disabled})}
    >
      <Icon name={props.name} size={size} color={props.color} />
    </TouchableOpacity>
  )
}

DSIcon.defaultProps = {
  color: Colors.black,
  size: 'small', // 'small', 'medium' , 'large', 'extra-large',
  name: 'eye'
}

DSIcon.propTypes = {
  onLongPress: PropTypes.func,
  onPress: PropTypes.func,
  onPressIn: PropTypes.func,
  onPressOut: PropTypes.func,
  name: PropTypes.string,
  size: PropTypes.oneOf(['xxx-small', 'extra-small', 'small', 'medium', 'large', 'extra-large']),
  color: PropTypes.string
}

export default DSIcon
