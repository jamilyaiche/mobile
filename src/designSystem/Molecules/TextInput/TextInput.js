import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import DSTextInput from '../../Atoms/DSTextInput/DSTextInput'
import DSLabel from '../../Atoms/DSLabel/DSLabel'
import Colors from '../../../assets/styles/Colors'
import Style from '../../../assets/styles/Style'
import DSIcon from '../../Atoms/DSIcon/DSIcon'

const TextInput = (props) => {
  const [focused, setFocused] = React.useState(false)
  const [passwordHidden, setPasswordHidden] = React.useState(true)
  

  const displayStyle1 = () => {
    return (
      <View>
        {displayLabel()}
        <DSTextInput
          {...(props.inputRef !== undefined && { inputRef: props.inputRef })}
          {...(props.value !== undefined && { value: props.value })}
          value={props.value}

          onFocus={() => {
            setFocused(true)
            if (props.onFocus !== undefined) {
              props.onFocus()
            }
          }}
          onBlur={() => {
            setFocused(false)
            if (props.onBlur) {
              props.onBlur()
            }
          }}
          placeholder={props.placeholder}
          autoCapitalize={props.autoCapitalize}
          autoCorrect={props.autoCorrect}
          editable={props.editable}
          keyboardType={props.keyboardType}
          {...(props.maxLength !== undefined && { maxLength: props.maxLength })}
          multiline={props.multiline}
          numberOfLines={props.numberOfLines}
          {...(props.onChange !== undefined && { onChange: props.onChange })}
          {...(props.onChangeText !== undefined && { onChangeText: props.onChangeText })}
          {...(props.onContentSizeChange !== undefined && { onContentSizeChange: props.onContentSizeChange })}
          {...(props.onEndEditing !== undefined && { onEndEditing: props.onEndEditing })}
          {...(props.onKeyPress !== undefined && { onKeyPress: props.onKeyPress })}
          {...(props.onLayout !== undefined && { onLayout: props.onLayout })}
          {...(props.onScroll !== undefined && { onScroll: props.onScroll })}
          {...(props.onSelectionChange !== undefined && { onSelectionChange: props.onSelectionChange })}
          {...(props.onSubmitEditing !== undefined && { onSubmitEditing: props.onSubmitEditing })}
          placeholderTextColor={props.placeholderTextColor || Colors.black(0.3)}
          returnKeyType={props.returnKeyType}
          secureTextEntry={(props.type === 'password' && passwordHidden)}
          {...(props.selectionColor !== undefined && { selectionColor: props.selectionColor })}
          selectTextOnFocus={props.selectTextOnFocus}
          style={[
            Style.textInput,
            props.leftIconName && {paddingLeft: 40} , 
            props.multiline && Style.textInputMultiline ,
            focused && !props.isInvalid && props.withBorderBottem && Style.borderBottom,
            props.rightIconName && {paddingRight: 40},
            props.isInvalid && Style.boderError,
            
          ]}

        />
        {displayRightIcon()}
        {displayErrorMessageStyle1()}
        {displayLeftIcon()}
      </View>
    )
  }

  const displayLabel= () => {
      return (
        <DSLabel
          {...(props.onLabelLayout !== undefined && { onLayout: props.onLabelLayout })}
          {...(props.onLabelLongPress !== undefined && { onLongPress: props.onLabelLongPress })}
          {...(props.onLabelPress !== undefined && { onPress: props.onLabelPress })}
          {...(props.onLabelTextLayout !== undefined && { onTextLayout: props.onLabelTextLayout })}
          selectable={props.selectable}
          style={Style.textInputLabel}
        >
          {props.label}
        </DSLabel>
      )
  }
  const displayLeftIcon=()=>{
    if(props.leftIconName){
    return (
      <DSIcon
        onPress={props.onLeftIconPress ? props.onLeftIconPress : null}
        {...(props.onLeftIconPressIn !== undefined && { onPressIn: props.onLeftIconPressIn })}
        {...(props.onLeftIconPressOut !== undefined && { onPressOut: props.onLeftIconPressOut })}
        {...(props.onLeftIconLongPress !== undefined && { onLongPress: props.onLeftIconLongPress })}
        style={[Style.TextInputLeftIcon,!props.label && {top:32},props.leftIconStyle ]}
        name={props.leftIconName}
        color={props.leftIconColor} 
        size={props.leftIconSize}
      />
    )
    }
  }
  const displayRightIcon = () => {
    if (props.rightIconName) {
      return (
        <DSIcon
          onPress={props.onRightIconPress ? props.onRightIconPress : null}
          {...(props.onRightIconPressIn !== undefined && { onPressIn: props.onRightIconPressIn })}
          {...(props.onRightIconPressOut !== undefined && { onPressOut: props.onRightIconPressOut })}
          {...(props.onRightIconLongPress !== undefined && { onLongPress: props.onRightIconLongPress })}
          style={[Style.TextInputRightIcon,!props.label && {top:38},props.rightIconStyle ]}
          name={props.rightIconName}
          color={props.rightIconColor} 
          size={props.rightIconSize}
        />
      )
    }
    return null
  }
  const displayErrorMessageStyle1 = () => {
    if (props.isInvalid) {
      return (
        <DSLabel style={Style.textInputErrorMessageStyle1}>{props.errorMessage}</DSLabel>
      )
    }
  }
    return displayStyle1()
}
TextInput.defaultProps = {
  leftIconSize:"small",
  leftIconStyle:{},
  leftIconColor:Colors.black(),
  rightIconSize:'xxx-small',
  rightIconColor:Colors.black(),
  rightIconStyle:{}
}
TextInput.propTypes = {
  autoCapitalize: PropTypes.any,
  autoCorrect: PropTypes.any,
  editable: PropTypes.any,
  errorMessage: PropTypes.any,
  inputRef: PropTypes.any,
  isInvalid: PropTypes.any,
  keyboardType: PropTypes.any,
  label: PropTypes.any,
  leftIconName: PropTypes.shape({
    paddingLeft: PropTypes.number
  }),
  maxLength: PropTypes.any,
  multiline: PropTypes.any,
  numberOfLines: PropTypes.any,
  onBlur: PropTypes.func,
  onChange: PropTypes.any,
  onChangeText: PropTypes.any,
  onContentSizeChange: PropTypes.any,
  onEndEditing: PropTypes.any,
  onFocus: PropTypes.func,
  onKeyPress: PropTypes.any,
  onLabelLayout: PropTypes.any,
  onLabelLongPress: PropTypes.any,
  onLabelPress: PropTypes.any,
  onLabelTextLayout: PropTypes.any,
  onLayout: PropTypes.any,
  onRightIconLongPress: PropTypes.any,
  onRightIconPress: PropTypes.any,
  onRightIconPressIn: PropTypes.any,
  onRightIconPressOut: PropTypes.any,
  onScroll: PropTypes.any,
  onSelectionChange: PropTypes.any,
  onSubmitEditing: PropTypes.any,
  placeholder: PropTypes.any,
  placeholderTextColor: PropTypes.any,
  returnKeyType: PropTypes.any,
  rightIconName: PropTypes.shape({
    paddingRight: PropTypes.number
  }),
  selectTextOnFocus: PropTypes.any,
  selectable: PropTypes.any,
  selectionColor: PropTypes.any,
  type: PropTypes.string,
  value: PropTypes.any
}

export default TextInput
