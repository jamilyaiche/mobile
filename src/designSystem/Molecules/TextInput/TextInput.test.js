import React from 'react';
import renderer from 'react-test-renderer'
import TextInput from "./TextInput"
import DSTextInput from "../../Atoms/DSTextInput/DSTextInput"
import DSIcon from "../../Atoms/DSIcon/DSIcon"
import { shallow } from 'enzyme'

describe("TextInput tests",()=>{
    it('test render text input snapshot',()=>{
        const wrapper=renderer.create(<TextInput value="test" lable="Label" placeholder="Placeholder" />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })

    it('test text input value',()=>{
        const wrapper=shallow(<TextInput value="test" lable="Label" placeholder="Placeholder" />)
        expect(wrapper.find(DSTextInput).first().props().value===`test`)
        .toEqual(true)
    })

    it('test render password text input snapshot',()=>{
        const wrapper=renderer.create(<TextInput type='password' value="" lable="Label" placeholder="Placeholder" />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })

    it('test password type',()=>{
        const wrapper=shallow(<TextInput type='password' value="" lable="Label" placeholder="Placeholder" />)
        expect(wrapper.find(DSTextInput).first().props().secureTextEntry===true)
        .toEqual(true)
    })

    it('test render text input width right icon camera snapshot',()=>{
        const wrapper=renderer.create(<TextInput rightIcon="camera" value="" lable="Label" placeholder="Placeholder" />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })




    it('test render multiliane text input snapshot',()=>{
        const wrapper=renderer.create(<TextInput multiline value="" lable="Label" placeholder="Placeholder" />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })
    it('test render text input with error message snapshot',()=>{
        const wrapper=renderer.create(<TextInput errorMessage="error" value="" lable="Label" placeholder="Placeholder" />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })
    it('test all props Text Left input snapshot',()=>{
        const wrapper=renderer.create(
        <TextInput 
            autoCorrect={true}
            defaultValue='test'
            editable={false}
            errorMessage='test error msg'
            isInvalid='true'
            keyboardType='phone-pad'
            label='test'
            maxLength={10}
            multiline={true}
            numberOfLines={1}
            placeholder='test'
            placeholderTextColor='red'
            rightIcon='x'
            selectTextOnFocus={true}
            showIconOnleft={true}
            styleType='default'
            type='password'
            valididtyCriterias={[]}
            value='test'
             />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })
    it('test all props Right Icon Text input snapshot',()=>{
        const wrapper=renderer.create(
        <TextInput 
            autoCorrect={true}
            defaultValue='test'
            editable={false}
            errorMessage='test error msg'
            isInvalid='true'
            keyboardType='phone-pad'
            label='test'
            maxLength={10}
            multiline={true}
            numberOfLines={1}
            placeholder='test'
            placeholderTextColor='red'
            rightIcon='x'
            selectTextOnFocus={true}
            showIconOnleft={false}
            styleType='default'
            type='password'
            value='test'
            valididtyCriterias={[{text:"> 8 caractères" ,valid:false},{text:"1 majuscule" ,valid:true},{text:"> 1 caractère spécial" ,valid:false}]}
             />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })
    
})