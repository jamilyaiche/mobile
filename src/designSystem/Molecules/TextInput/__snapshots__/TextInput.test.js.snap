// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput tests test all props Right Icon Text input snapshot 1`] = `
<View>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Object {
        "color": "rgba(26, 26, 26, 1)",
        "fontFamily": "Roboto-Bold",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 1,
        "lineHeight": 21,
      }
    }
  >
    test
  </Text>
  <TextInput
    allowFontScaling={false}
    autoCapitalize="sentences"
    autoCorrect={true}
    defaultValue=""
    editable={false}
    inputAccessoryViewID="id"
    keyboardType="phone-pad"
    maxLength={10}
    multiline={true}
    numberOfLines={1}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="test"
    placeholderTextColor="red"
    rejectResponderTermination={true}
    returnKeyType="done"
    secureTextEntry={true}
    selectTextOnFocus={true}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(229, 229, 229, 0.3)",
          "borderRadius": 5,
          "color": "rgba(26, 26, 26, 1)",
          "fontFamily": "Roboto-Bold",
          "fontSize": 12,
          "fontWeight": "700",
          "height": 56,
          "letterSpacing": 1,
          "lineHeight": 15,
          "paddingLeft": 16,
          "paddingRight": 16,
          "width": "100%",
        },
        undefined,
        Object {
          "height": 150,
          "paddingTop": 20,
          "textAlignVertical": "top",
        },
        false,
        undefined,
        Object {
          "borderColor": "rgba(241, 39, 39, 1)",
          "borderWidth": 1,
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="test"
  />
  <RCTInputAccessoryView
    nativeID="id"
    style={
      Array [
        undefined,
        Object {
          "position": "absolute",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </RCTInputAccessoryView>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Object {
        "color": "rgba(241, 39, 39, 1)",
        "fontFamily": "Roboto-Bold",
        "fontSize": 12,
        "lineHeight": 18,
        "marginTop": 10,
        "width": "100%",
      }
    }
  >
    test error msg
  </Text>
</View>
`;

exports[`TextInput tests test all props Text Left input snapshot 1`] = `
<View>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Object {
        "color": "rgba(26, 26, 26, 1)",
        "fontFamily": "Roboto-Bold",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 1,
        "lineHeight": 21,
      }
    }
  >
    test
  </Text>
  <TextInput
    allowFontScaling={false}
    autoCapitalize="sentences"
    autoCorrect={true}
    defaultValue=""
    editable={false}
    inputAccessoryViewID="id"
    keyboardType="phone-pad"
    maxLength={10}
    multiline={true}
    numberOfLines={1}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="test"
    placeholderTextColor="red"
    rejectResponderTermination={true}
    returnKeyType="done"
    secureTextEntry={true}
    selectTextOnFocus={true}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(229, 229, 229, 0.3)",
          "borderRadius": 5,
          "color": "rgba(26, 26, 26, 1)",
          "fontFamily": "Roboto-Bold",
          "fontSize": 12,
          "fontWeight": "700",
          "height": 56,
          "letterSpacing": 1,
          "lineHeight": 15,
          "paddingLeft": 16,
          "paddingRight": 16,
          "width": "100%",
        },
        undefined,
        Object {
          "height": 150,
          "paddingTop": 20,
          "textAlignVertical": "top",
        },
        false,
        undefined,
        Object {
          "borderColor": "rgba(241, 39, 39, 1)",
          "borderWidth": 1,
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="test"
  />
  <RCTInputAccessoryView
    nativeID="id"
    style={
      Array [
        undefined,
        Object {
          "position": "absolute",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </RCTInputAccessoryView>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Object {
        "color": "rgba(241, 39, 39, 1)",
        "fontFamily": "Roboto-Bold",
        "fontSize": 12,
        "lineHeight": 18,
        "marginTop": 10,
        "width": "100%",
      }
    }
  >
    test error msg
  </Text>
</View>
`;

exports[`TextInput tests test render multiliane text input snapshot 1`] = `
<View>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Object {
        "color": "rgba(26, 26, 26, 1)",
        "fontFamily": "Roboto-Bold",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 1,
        "lineHeight": 21,
      }
    }
  />
  <TextInput
    allowFontScaling={false}
    autoCapitalize="sentences"
    autoCorrect={false}
    defaultValue=""
    editable={true}
    inputAccessoryViewID="id"
    keyboardType="default"
    multiline={true}
    numberOfLines={5}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Placeholder"
    rejectResponderTermination={true}
    returnKeyType="done"
    secureTextEntry={false}
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(229, 229, 229, 0.3)",
          "borderRadius": 5,
          "color": "rgba(26, 26, 26, 1)",
          "fontFamily": "Roboto-Bold",
          "fontSize": 12,
          "fontWeight": "700",
          "height": 56,
          "letterSpacing": 1,
          "lineHeight": 15,
          "paddingLeft": 16,
          "paddingRight": 16,
          "width": "100%",
        },
        undefined,
        Object {
          "height": 150,
          "paddingTop": 20,
          "textAlignVertical": "top",
        },
        false,
        undefined,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <RCTInputAccessoryView
    nativeID="id"
    style={
      Array [
        undefined,
        Object {
          "position": "absolute",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </RCTInputAccessoryView>
</View>
`;

exports[`TextInput tests test render password text input snapshot 1`] = `
<View>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Object {
        "color": "rgba(26, 26, 26, 1)",
        "fontFamily": "Roboto-Bold",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 1,
        "lineHeight": 21,
      }
    }
  />
  <TextInput
    allowFontScaling={false}
    autoCapitalize="sentences"
    autoCorrect={false}
    defaultValue=""
    editable={true}
    inputAccessoryViewID="id"
    keyboardType="default"
    multiline={false}
    numberOfLines={5}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Placeholder"
    rejectResponderTermination={true}
    returnKeyType="done"
    secureTextEntry={true}
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(229, 229, 229, 0.3)",
          "borderRadius": 5,
          "color": "rgba(26, 26, 26, 1)",
          "fontFamily": "Roboto-Bold",
          "fontSize": 12,
          "fontWeight": "700",
          "height": 56,
          "letterSpacing": 1,
          "lineHeight": 15,
          "paddingLeft": 16,
          "paddingRight": 16,
          "width": "100%",
        },
        undefined,
        undefined,
        false,
        undefined,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <RCTInputAccessoryView
    nativeID="id"
    style={
      Array [
        undefined,
        Object {
          "position": "absolute",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </RCTInputAccessoryView>
</View>
`;

exports[`TextInput tests test render text input snapshot 1`] = `
<View>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Object {
        "color": "rgba(26, 26, 26, 1)",
        "fontFamily": "Roboto-Bold",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 1,
        "lineHeight": 21,
      }
    }
  />
  <TextInput
    allowFontScaling={false}
    autoCapitalize="sentences"
    autoCorrect={false}
    defaultValue=""
    editable={true}
    inputAccessoryViewID="id"
    keyboardType="default"
    multiline={false}
    numberOfLines={5}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Placeholder"
    rejectResponderTermination={true}
    returnKeyType="done"
    secureTextEntry={false}
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(229, 229, 229, 0.3)",
          "borderRadius": 5,
          "color": "rgba(26, 26, 26, 1)",
          "fontFamily": "Roboto-Bold",
          "fontSize": 12,
          "fontWeight": "700",
          "height": 56,
          "letterSpacing": 1,
          "lineHeight": 15,
          "paddingLeft": 16,
          "paddingRight": 16,
          "width": "100%",
        },
        undefined,
        undefined,
        false,
        undefined,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value="test"
  />
  <RCTInputAccessoryView
    nativeID="id"
    style={
      Array [
        undefined,
        Object {
          "position": "absolute",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </RCTInputAccessoryView>
</View>
`;

exports[`TextInput tests test render text input width right icon camera snapshot 1`] = `
<View>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Object {
        "color": "rgba(26, 26, 26, 1)",
        "fontFamily": "Roboto-Bold",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 1,
        "lineHeight": 21,
      }
    }
  />
  <TextInput
    allowFontScaling={false}
    autoCapitalize="sentences"
    autoCorrect={false}
    defaultValue=""
    editable={true}
    inputAccessoryViewID="id"
    keyboardType="default"
    multiline={false}
    numberOfLines={5}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Placeholder"
    rejectResponderTermination={true}
    returnKeyType="done"
    secureTextEntry={false}
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(229, 229, 229, 0.3)",
          "borderRadius": 5,
          "color": "rgba(26, 26, 26, 1)",
          "fontFamily": "Roboto-Bold",
          "fontSize": 12,
          "fontWeight": "700",
          "height": 56,
          "letterSpacing": 1,
          "lineHeight": 15,
          "paddingLeft": 16,
          "paddingRight": 16,
          "width": "100%",
        },
        undefined,
        undefined,
        false,
        undefined,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <RCTInputAccessoryView
    nativeID="id"
    style={
      Array [
        undefined,
        Object {
          "position": "absolute",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </RCTInputAccessoryView>
</View>
`;

exports[`TextInput tests test render text input with error message snapshot 1`] = `
<View>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Object {
        "color": "rgba(26, 26, 26, 1)",
        "fontFamily": "Roboto-Bold",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 1,
        "lineHeight": 21,
      }
    }
  />
  <TextInput
    allowFontScaling={false}
    autoCapitalize="sentences"
    autoCorrect={false}
    defaultValue=""
    editable={true}
    inputAccessoryViewID="id"
    keyboardType="default"
    multiline={false}
    numberOfLines={5}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Placeholder"
    rejectResponderTermination={true}
    returnKeyType="done"
    secureTextEntry={false}
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(229, 229, 229, 0.3)",
          "borderRadius": 5,
          "color": "rgba(26, 26, 26, 1)",
          "fontFamily": "Roboto-Bold",
          "fontSize": 12,
          "fontWeight": "700",
          "height": 56,
          "letterSpacing": 1,
          "lineHeight": 15,
          "paddingLeft": 16,
          "paddingRight": 16,
          "width": "100%",
        },
        undefined,
        undefined,
        false,
        undefined,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <RCTInputAccessoryView
    nativeID="id"
    style={
      Array [
        undefined,
        Object {
          "position": "absolute",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </RCTInputAccessoryView>
</View>
`;
